File: test_line_chart.py
import unittest
from selenium import webdriver
from selenium.webdriver.chrome.service import Service
from line_chart_page import LineChartPage

class TestLineChart(unittest.TestCase):

    def setUp(self):
        service = Service('C:/Automation/chromedriver.exe')
        self.driver = webdriver.Chrome(service=service)
        self.driver.get('https://inerg-test.web.app/')

    def test_select_state_and_print_chart_values(self):
        line_chart_page = LineChartPage(self.driver)
        line_chart_page.select_state_and_wait_for_chart('Kerala')  # Update with actual state
        line_chart_page.print_text_values()  # Print the text values

    def tearDown(self):
        self.driver.quit()

if __name__ == "__main__":
    unittest.main()

File: test/java/com/example/TestLineChart.java
package com.example;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;

public class TestLineChart {

    private WebDriver driver;

    @Before
    public void setUp() {
        System.setProperty("webdriver.chrome.driver", "C:/Automation/chromedriver.exe");
        ChromeOptions options = new ChromeOptions();
        driver = new ChromeDriver(options);
        driver.get("https://inerg-test.web.app/");
    }

    @Test
    public void testSelectStateAndPrintChartValues() {
        LineChartPage lineChartPage = new LineChartPage(driver);
        lineChartPage.selectStateAndWaitForChart("Kerala");  // Update with actual state
        lineChartPage.printTextValues();  // Print the text values
    }

    @After
    public void tearDown() {
        if (driver != null) {
            driver.quit();
        }
    }
}